### PR TITLE
fix: RunPdf.Exec threw ArgumentNullException resolving path to wkhtmltopdf

### LINKED ISSUE #8348: [Bug] Docset.Build throws ArgumentNullException if `pdf` is set in docfx.json
**Describe the bug**
At https://github.com/dotnet/docfx/blob/main/src/Microsoft.DocAsCode.App/RunPdf.cs#L18, `config.BaseDirectory` is passed as the first argument to Path.Combine but it is always null, so it throws ArgumentNullException.

Based on what RunBuild.cs is doing, I believe the fix would be to move https://github.com/dotnet/docfx/blob/main/src/Microsoft.DocAsCode.App/RunPdf.cs#L32-L34 to the top of the method, and pass `EnvironmentContext.BaseDirectory` to Path.Combine instead.

**To Reproduce**
1. Call Docset.Build, passing the path to a docfx.json defining PDF output.

**Expected behavior**
A PDF to be generated.

**Context (please complete the following information):**
- OS: Ubuntu 22.04 / WSL2
- Docfx version: 2.60.0
- .NET version: .NET 6.0

- `docfx.json` config

```json
{
  "build": {
    "content": [
      {
        "src": "docs",
        "files": [
          "platform/**",
          "frontend/**",
          "content/**",
          "minutes/**",
          "index.md",
          "toc.yml"
        ],
        "exclude": ["**/.attachments/**"]
      },
      {
        "files": [
          "{frontend,platform}/README.md",
          "platform/*/README.md",
          "platform/*/docs/**.{md,yml}",
          "frontend/{apis,lib,web,desktop}/*/README.md",
          "frontend/{apis,lib,web,desktop}/*/docs/**.{md,yml}"
        ],
        "exclude": ["**/ci/**", "**/node_modules/**", "**/.attachments/**"]
      }
    ],
    "resource": [
      {
        "src": "docs",
        "files": ["content/.attachments/**", "content/**/.attachments/**", "minutes/.attachments/**", "minutes/**/.attachments/**"]
      },
      {
        "files": [
          "frontend/{apis,lib,web,desktop}/*/docs/.attachments/**",
          "platform/*/docs/.attachments/**",
          "frontend/{apis,lib,web,desktop}/*/docs/**/.attachments/**",
          "platform/*/docs/**/.attachments/**"
        ]
      }
    ],
    "dest": "docs/_site",
    "template": [
      "docs/template"
    ],
    "globalMetadata": {
      "_enableSearch": true,
      "_disableBreadcrumb": false
    },
    "fileMetadata": {
      "_disableContribution": {
        "**/swagger.json": true
      }
    },
    "globalMetadataFiles": [],
    "fileMetadataFiles": [],
    "postProcessors": [],
    "markdownEngineName": "markdig",
    "markdownEngineProperties": {
      "plantUml.renderingMode": "remote",
      "plantUml.remoteUrl": "http://localhost:8080/",
      "markdigExtensions": [
          "mathematics",
          "diagrams"
      ]
    },
    "noLangKeyword": false,
    "keepFileLink": false,
    "cleanupCacheHistory": true,
    "disableGitFeatures": false
  },
  "pdf": {
    "content": [
      {
        "src": "docs",
        "files": ["platform/**", "frontend/**", "minutes/**", "content/**", "index.md"],
        "exclude": ["**/.attachments/**"]
      },
      {
        "files": [
          "{frontend,platform}/README.md",
          "platform/*/README.md",
          "platform/*/docs/**.{md,yml}",
          "frontend/{apis,lib,web,desktop}/*/README.md",
          "frontend/{apis,lib,web,desktop}/*/docs/**.{md,yml}"
        ],
        "exclude": ["**/ci/**", "**/node_modules/**", "**/.attachments/**"]
      },
      {
        "src": "docs/pdf",
        "files": ["toc.yml"]
      }
    ],
    "resource": [
      {
        "src": "docs",
        "files": ["content/.attachments/**", "content/**/.attachments/**", "minutes/.attachments/**", "minutes/**/.attachments/**"]
      },
      {
        "files": [
          "frontend/{apis,lib,web,desktop}/*/docs/.attachments/**",
          "platform/*/docs/.attachments/**",
          "frontend/{apis,lib,web,desktop}/*/docs/**/.attachments/**",
          "platform/*/docs/**/.attachments/**"
        ]
      }
    ],
    "wkhtmltopdf": {
      "filePath": "/usr/bin/wkhtmltopdf",
      "additionalArguments": "--enable-local-file-access"
    },
    "dest": "docs/_site_pdf",
    "template": [
      "pdf.default"
    ],
    "globalMetadata": {
      "_enableSearch": false,
      "_disableBreadcrumb": false
    },
    "fileMetadata": {
      "_disableContribution": {
        "**/swagger.json": true
      }
    },
    "globalMetadataFiles": [],
    "fileMetadataFiles": [],
    "postProcessors": [],
    "markdownEngineName": "markdig",
    "markdownEngineProperties": {
      "plantUml.renderingMode": "remote",
      "plantUml.remoteUrl": "http://localhost:8080/",
      "markdigExtensions": [
          "mathematics",
          "diagrams"
      ]
    },
    "noLangKeyword": false,
    "keepFileLink": false,
    "cleanupCacheHistory": true,
    "disableGitFeatures": false
  }
}

```

- Exceptions

```
System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.IO.Path.Combine(String path1, String path2)
   at Microsoft.DocAsCode.RunPdf.Exec(PdfJsonConfig config)
   at Microsoft.DocAsCode.Docset.Build(String configPath)
   at Program.<Main>$(String[] args) in /home/chrisv/repos/redacted/docs-tools/docs-builder/Program.cs:line 3
```

- .NET info

```
❯ dotnet --info
.NET SDK:
 Version:   7.0.102
 Commit:    4bbdd14480

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-x64
 Base Path:   /usr/share/dotnet/sdk/7.0.102/

Host:
  Version:      7.0.2
  Architecture: x64
  Commit:       d037e070eb

.NET SDKs installed:
  6.0.405 [/usr/share/dotnet/sdk]
  7.0.102 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.13 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.13 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/usr/lib/dotnet/dotnet6-6.0.109]

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

**Additional context**
contents of `Program.cs`:
```csharp
// See https://aka.ms/new-console-template for more information

await Microsoft.DocAsCode.Docset.Build("docfx.json");
```

### PR DESCRIPTION
**Pull request by chrisvanderpennen**
Fixes #8348

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=8350) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=8350) before we can accept your contribution.<br/><br/>:x: chrisvanderpennen [sign now](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=8350)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docfx?pullRequest=8350) it.</sub>

File: src/Microsoft.DocAsCode.App/RunPdf.cs [hunk 1]
```diff
@@ -13,9 +13,23 @@ namespace Microsoft.DocAsCode
...
+        }
+
         public static void Exec(PdfJsonConfig config)
         {
-            var wkhtmltopdfFilePath = Path.Combine(config.BaseDirectory, config.Wkhtmltopdf?.FilePath);
+            EnvironmentContext.SetBaseDirectory(Path.GetFullPath(string.IsNullOrEmpty(config.BaseDirectory) ? Directory.GetCurrentDirectory() : config.BaseDirectory));
+            // TODO: remove BaseDirectory from Config, it may cause potential issue when abused
+            var baseDirectory = EnvironmentContext.BaseDirectory;
+
+            var wkhtmltopdfFilePath = GetFullFilePath(config.Wkhtmltopdf?.FilePath, baseDirectory);
```
Comment (yufeih) [inline src/Microsoft.DocAsCode.App/RunPdf.cs:32]: The `IsPathRooted` check is in the helper method should have the same effect as `Path.Combine` so I was removing the helper method, turns out causing `null` handling problems. Consider:

```suggestion
            var wkhtmltopdfFilePath = config.Wkhtmltopdf?.FilePath is null ? null : Path.Combine(config.Wkhtmltopdf.FilePath, baseDirectory);
```

**Review by yufeih — APPROVED** [review]

**Comment by yufeih** [comment]
> [CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=8350) Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=8350) before we can accept your contribution.❌ chrisvanderpennen [sign now](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=8350)You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docfx?pullRequest=8350) it.

This needs to be signed before merging the PR.

File: src/Microsoft.DocAsCode.App/RunPdf.cs (hunk 1 above)
Comment (chrisvanderpennen) [reply src/Microsoft.DocAsCode.App/RunPdf.cs:32]: Sure, I'll remove the helper as well.

**Comment by yufeih** [comment]
Thank you!

**Pull request merged**